pkg/pq/priority_queue.go

```diff
@@ -51,16 +51,38 @@ func (pq *PriorityQueue) Len() int {
 }
 
 func (pq *PriorityQueue) Release(value Value) {
-	queue := &internalQueue{}
-	heap.Init(queue)
+	idx := pq.findIdx(func(item *Item) bool {
+		return item.value == value
+	})
 
-	for _, item := range *pq.queue {
-		if item.value != value {
-			heap.Push(queue, item)
-		}
+	if idx == -1 {
+		return
 	}
 
-	pq.queue = queue
+	// Item to replace
+	lastIdx := (*pq.queue).Len() - 1
+
+	lastItem := (*pq.queue)[lastIdx]
+	lastItem.index = idx
+
+	(*pq.queue)[idx] = lastItem
+	heap.Fix(pq.queue, idx)
+
+	(*pq.queue)[lastIdx] = nil
+	(*pq.queue) = (*pq.queue)[:lastIdx]
+
+	if (*pq.queue).Len() == 0 {
+		pq.queue = &internalQueue{}
+	}
```

Comment:
In the `Release` function, how about using the `peek` function to implement this? 🤔
For example, 
```go
idx := pq.findIdx(func(item *Item) bool {
  return item.value == value
})

lastItem := pq.queue.Peek().(*Item)
if idx == -1 || (*pq.queue).Len() == 0 {
  return
}

(*pq.queue)[idx] = lastItem
heap.Fix(pq.queue, idx)
```